ios/FluentUI/Controls/IndeterminateProgressBarView.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+/// Indeterminate progress bar. Since iOS's UIprogressView is always determinate, this is created for handling indeterminate scenarios.
+@objc(MSFIndeterminateProgressBarView)
+open class IndeterminateProgressBarView: UIView {
+
+    /// The progress bar view should be hidden when animation stops if set to true
+    @objc open var hidesWhenStopped: Bool = true
+
+    open override var intrinsicContentSize: CGSize {
+        return CGSize(width: UIView.noIntrinsicMetric, height: frame.height)
+    }
+
+    open override func sizeThatFits(_ size: CGSize) -> CGSize {
+        return CGSize(width: size.width, height: Constants.progressBarHeight)
+    }
+
+    @objc public override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.frame.size.height = Constants.progressBarHeight
+        autoresizingMask = .flexibleWidth
+        clipsToBounds = true
+    }
+
+    required public init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    open override func layoutSubviews() {
+        super.layoutSubviews()
+
+        trackLayer.removeFromSuperlayer()
+        trackLayer.frame = CGRect(x: 0, y: 0, width: self.frame.width, height: self.frame.size.height)
+        layer.addSublayer(trackLayer)
+
+        addGradientLayer()
+        animationGroup = addAnimation()
+        startAnimating()
+    }
+
+    open override func didMoveToWindow() {
+        super.didMoveToWindow()
+        updateColor()
+    }
+
+    @objc open func startAnimating() {
+        if isAnimating {
+            return
+        }
+        isAnimating = true
+
+        isHidden = false
+        gradientLayer.add(animationGroup, forKey: Constants.animationKey)
+    }
+
+    @objc open func stopAnimating() {
+        if !isAnimating {
```

Comment:
Can you use guard..else here as the best practice standards outline?

```
    guard isAnimating else {
        return
    }
```

https://github.com/microsoft/swift-guide/blob/gh-pages/EarlyReturn.md